### PR TITLE
add solution of first part

### PR DESCRIPTION
**Pull request by ssaviak**
[DEMO LINK](https://ssaviak.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

**Review by OleksandrRezanov — APPROVED** [review]
Terrific work!

File: src/App.tsx [hunk 1]
```diff
@@ -1,204 +1,58 @@
...
+  const {
+    setErrorMessage,
+    todos,
+    setTodos,
+  } = useContext(TodoContext);
+
+  const handleOverLoad = () => {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
```
Comment (olya-shyian) [inline src/App.tsx:24]: its better to keep logic about error (setTimeout) inside `Error` component

you can use `useEffect`, if you have some `ErrorMessage` then reset it after 3000

File: src/App.tsx [hunk 2]
```diff
@@ -1,204 +1,58 @@
...
+  };
+
+  useEffect(() => {
+    getTodos(84)
```
Comment (olya-shyian) [inline src/App.tsx:28]: what `84`  means?

File: src/App.tsx [hunk 3]
```diff
@@ -1,204 +1,58 @@
...
+      .then((res) => setTodos(res))
```
Comment (olya-shyian) [inline src/App.tsx:29]: try to avoid words like `res`, `func`, `el` etc

File: src/components/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        </a>
+
+        <a
+          href="#/completed"
+          className={classNames(
+            'filter__link',
+            { selected: selectedFilter === 'completed' },
+          )}
+          data-cy="FilterLinkCompleted"
+          onClick={handleClick('completed')}
```
Comment (olya-shyian) [inline src/components/Footer.tsx:56]: please create enum for these filter types like `All`, `Active`, `Completed`

File: src/components/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import classNames from 'classnames';
+
+import { TodoContext } from '../contexts/TodoContext';
+import { TodoStatus } from '../types/TodoStatus';
+
+export const Footer: React.FC = () => {
+  const { todos, setFilters } = useContext(TodoContext);
+  const [selectedFilter, setSelectedFilter] = useState('all');
+
+  const handleClick = (value: TodoStatus) => () => {
```
Comment (olya-shyian) [inline src/components/Footer.tsx:11]: please give more clear name

File: src/components/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+export const Header: React.FC = () => {
+  return (
```
Comment (olya-shyian) [inline src/components/Header.tsx:2]: when you don't have any logic inside component and just return jsx then you can delete the word `return`

and just return jsx in round brackets

File: src/components/TodoList.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    filteredTodos,
+    setTodos,
+    setErrorMessage,
+  } = useContext(TodoContext);
+
+  const handleCatch = () => {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
```
Comment (olya-shyian) [inline src/components/TodoList.tsx:18]: when you move this logic according to my previous comment you can avoid this code duplication with setTimeout

File: src/contexts/TodoContext.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      switch (filters.status) {
+        case 'uncompleted':
+          return !todo.completed;
+        case 'completed':
+          return todo.completed;
+        default:
+          return true;
+      }
+    }),
+  }), [todos, errorMessage, filters]);
```
Comment (olya-shyian) [inline src/contexts/TodoContext.tsx:53]: please move this filter logic to a separate function, and put it inside `utils` folder

File: src/types/ErrorMessage.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+export enum ErrorMessage {
+  failedLoad = 'Unable to load todos',
+  failedAddTodo = 'Unable to add a todo',
+  failedDeleteTodo = 'Unable to delete a todo',
+  failedUpdateTodo = 'Unable to update a todo',
+  emptyTitle = 'Title should not be empty',
```
Comment (olya-shyian) [inline src/types/ErrorMessage.ts:6]: great solution!
please just use first UpperCase letter like `FailedLoad`

File: src/components/TodoList.tsx [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+  const handleDeleteTodo = (todoId: number) => {
+    deleteTodo(todoId)
```
Comment (olya-shyian) [inline src/components/TodoList.tsx:21]: wow, you have already implemened `delete` functionality)
cool!)

**Review by olya-shyian — APPROVED** [review]
Super work!✨
I'll approve. In order to save your time you can do updates according to my comments in the next task

**Pull request closed without merging**